Framework/PythonInterface/plugins/algorithms/HB2AReduce.py

```diff
@@ -81,11 +117,44 @@ def validateInputs(self):
                 issues["Filename"] = 'Must specify either Filename or IPTS AND ScanNumbers'
 
             if self.getProperty("Exp").value == Property.EMPTY_INT:
-                exp_list = sorted(e for e in os.listdir('/HFIR/HB2A/IPTS-{0}'.format(ipts)) if 'exp' in e)
-                if len(exp_list)>1:
-                    exps = ','.join(e.replace('exp','') for e in exp_list)
-                    issues["Exp"] = 'Multiple experiments found in IPTS-{}. You must set Exp to one of {}'.format(ipts, exps)
-
+                exp_list = sorted(e for e in os.listdir('/HFIR/HB2A/IPTS-{0}'.format(ipts))
+                                  if 'exp' in e)
+                if len(exp_list) > 1:
+                    exps = ','.join(e.replace('exp', '') for e in exp_list)
+                    issues[
+                        "Exp"] = 'Multiple experiments found in IPTS-{}. You must set Exp to one of {}'.format(
+                            ipts, exps)
+
+        # validate output format options
+        # Def_x    GSAS    XYE
+        #  2theta   Y       Y
+        #  others   N       Y
+        if self.getProperty("SaveData").value & (self.getProperty('OutputFormat').value == "GSAS"):
+            _defx = self.getProperty("DefX").value
+            if not _defx:  # DefX is given, check if defx == 2theta
```

Comment:
```suggestion
            if _defx:  # DefX is given, check if defx == 2theta
```

I think you got this backwards?

Reply:
default value of _defx is "", an empty string.  I have switched to explicit compare (_defx != "") to avoid the strange inconsistent implicit casting on the build server (the changes are being tested with the sibling PR to ornl-next).